Lib/ntpath.py

```diff
@@ -334,14 +334,14 @@ def isreserved(path):
     """Return true if the pathname is reserved by the system."""
     # Refer to "Naming Files, Paths, and Namespaces":
     # https://docs.microsoft.com/en-us/windows/win32/fileio/naming-a-file
-    path = os.fsdecode(splitroot(path)[2]).replace(altsep, sep)
-    return any(_isreservedname(name) for name in reversed(path.split(sep)))
+    path = os.fsdecode(splitroot(path)[2]).replace('/', '\\')
+    return any(_isreservedname(name) for name in reversed(path.split('\\')))
```

Comment:
How much does this specific change speed things up? It doesn't look to me like the kind of thing that would lead to a speedup, so I'm a little surprised here

Reply:
This was to ensure nothing can break if these constants are monkeypatched.
I didn't make a separate issue for this, because it's not part of the API contract. (See the conversation above).

Reply:
I saw the conversation above, and the change is small enough that I'm fine with the change staying in this PR if there's a good motivation, but I'm still confused about the motivation. Most things in Python can be monkeypatched, and that's not something we really go out of our way to prevent (though third-party _optional_ linters may complain about it). It's just part of the language.

Reply:
Yeah, but it's a bit different here as it's _documented_ that they are hardcoded so it therefore would be fine to modify them.

Reply:
> it's _documented_ that they are hardcoded so it therefore would be fine to modify them.

What do you mean, exactly? Just the comment at the top of the module? That comment is aimed at people working on the code. It doesn't count as external documentation for people using the code. As far as I know, we don't advertise anywhere in user-facing documentation that it's "fine to modify" these constants.

Reply:
Yeah, just the internal comment. How do you call that? Internally documented?

Reply:
You win, you're a reviewer and you'll block this pull request until I give in.

Reply:
> Yeah, just the internal comment. How do you call that? Internally documented?

I wouldn't really call it "documentation" at all, I'd just call it a comment :)

I don't think the intent of the comment is to say, "They're hardcoded, therefore it's fine to modify them". I think it's just noting the fact that the constants are very simply defined as literal strings in the source code rather than being obtained through some call to the operating system or whatever. So I don't really see a good reason to change anything on these lines :)